Gradient buttons with swiftlint

henrycity opened the pull request:
Add Swiftlint script before building the project. But need to install local first via Brew or Pod.
Has the same feature with gradient-buttons but following style guide by Realm team.

lexvdherik committed:
WIP see circle on line

lexvdherik committed:
Make circle move over line

danielcwilson committed:
Merge remote-tracking branch 'origin/make-circle-move' into gradient-buttons

danielcwilson committed:
WIP - Gradient on nearest button/beacon

Also had to fix the merge since make-circle-move wasnt up to date yet, some things are still missing but im on it dw

danielcwilson committed:
Remove some angry warnings, fix button pressing

At some point we started using the whole beacon object instead of just the coordinate but i didnt know this because merging is hard

danielcwilson committed:
Merge remote-tracking branch 'origin/map-improved' into gradient-buttons

danielcwilson committed:
Fix variable mismatch

This branch is really just dev2 but all nice and merged at this point

danielcwilson committed:
Merge remote-tracking branch 'origin/map-improved' into gradient-buttons

lexvdherik committed:
[WIP] make the circle update location

danielcwilson committed:
Merge remote-tracking branch 'origin/make-circle-move' into gradient-buttons

Things are looking up here in the gradient-buttons branch

Commit:
Add swiftlint and correct all the errors and warnings

vaguelycompetent commented:
https://github.com/realm/SwiftLint#installation

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -10,25 +10,32 @@ import CoreLocation
+    var circleShapeDrawn: Bool = false
+    var beacon1: CLBeacon!
+    var beacon2: CLBeacon!
+    var cgPoint: CGPoint!
```
What does `cgPoint` mean? What does this variable do?

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -92,54 +84,69 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
-                    beacon.button.backgroundColor = UIColor.blue
-                    nearestBeaconCoordinate = beacon.coordinate
+            beaconsArray = beacons
+            for myBeacon in beaconInfo {
```
Rename it `myBeacon` to `beacon` as I did in the dev branch.

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -151,31 +158,128 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
+            //Add the line to the layer
+            self.mapView.layer.addSublayer(self.lineShapeLayer)
+
+//            while true{
```
Remember to remove the comments if not needed anymore.

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -151,31 +158,128 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
+
+    func calcXY(firstBeacon: BeaconInfo, secondBeacon: BeaconInfo) -> CGPoint {
+
+        if beaconsArray.count <= 1 && cgPoint != nil {
```
`beaconsArray.count <= 1 && cgPoint != nil `. What does this mean? Make a new variable `let condition = beaconsArray.count <= 1 && cgPoint != nil`. It will be easier to understand what the condition is.

danielcwilson replied:
I've never seen creating a variable for a conditional being good practice but I agree that good comments on unclear conditionals is the way forward

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -151,31 +158,128 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
+        for beacon in beaconsArray where beacon.minor == secondBeacon.value {
+            beacon2 = beacon
+        }
+        if (beacon1 == nil || beacon2 == nil) && cgPoint != nil {
```
Add a comment or make a new variable to understand what the condition is.

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -151,31 +158,128 @@ class ViewController: UIViewController, CLLocationManagerDelegate {
+            return cgPoint
+        }
+
+        if (beacon1.accuracy == -1 || beacon2.accuracy == -1) && cgPoint != nil {
```
Add a comment or make a new variable to understand what the condition is.

henrycity reviewed (commented):
Need to write more declarative code.

danielcwilson reviewed (changes requested):
Do the stuff Henry said

danielcwilson committed:
Improved comments

Try to fix the issues with request #7 

I also condensed the buttonPress function cause yay cleaner code

danielcwilson committed:
Apparently whitespace isnt allowed

Lets fix that

henrycity commented on MobileDevProject/ViewController.swift:
```diff
@@ -10,32 +10,37 @@ import CoreLocation
-    var navigatingBeacon: BeaconInfo!
+    var beacon1: CLBeacon!  // these three variables are all used by calcXY
+    var beacon2: CLBeacon!  // stored globally to allow them to persist between calls
+    var cgPoint: CGPoint!   // this means we can better handle loss of signal
```
Maybe change the naming? The comment is still not clear

danielcwilson replied:
I don't really know how to make it any clearer than it is already, that can be down to @lexvdherik

danielcwilson committed:
Add a nice red gradient but with the right file

I fully messed that up

lexvdherik committed:
Add some comments and name changes

lexvdherik committed:
Add more comments
